main.nf

```diff
@@ -218,24 +301,129 @@ process get_software_versions {
     fastp -v 2> v_fastp.txt
     megahit --version > v_megahit.txt
     metabat2 -h 2> v_metabat.txt || true
+    NanoPlot --version > v_nanoplot.txt
+    filtlong --version > v_filtlong.txt
+    porechop --version > v_porechop.txt
+    NanoLyse --version > v_nanolyse.txt
+    spades.py --version > v_spades.txt
+    run_BUSCO.py --version > v_busco.txt
 
-    # fake checkm data setRoot so we can read checkm version number
-    chd=\$(readlink -f checkm_data)
-    printf "\$chd\\n\$chd\\n" | checkm data setRoot
-
-    checkm -h > v_checkm.txt
-    refinem -h > v_refinem.txt
     scrape_software_versions.py > software_versions_mqc.yaml
     """
 }
 
 
+/*
+ * Trim adapter sequences on long read nanopore files
+ */
+process porechop { 
+    tag "$id"
+        
+    input:
+    set id, lr, sr1, sr2 from files_all_raw
+    
+    output:
+    set id, file("${id}_porechop.fastq"), sr1, sr2 into files_porechop
+    set id, lr, val("raw") into files_nanoplot_raw
+    
+    script:
+    """
+    porechop -i ${lr} -t "${task.cpus}" -o ${id}_porechop.fastq
+    """
+}
+
+/*
+ * Remove reads mapping to the lambda genome.
+ * TODO: add lambda phage to igenomes.config?
+ */
+if (!params.keep_lambda) {
```

Comment:
I think you can use a `when` clause instead of the `if` here: [https://www.nextflow.io/docs/latest/process.html#when](https://www.nextflow.io/docs/latest/process.html#when)

Reply:
As far as I understand the when clause would also not forward any results, therefore the pipeline would just discard that info. This is fine for final results such as BUSCO, but not with intermediate steps like this one. If this step is skipped the files still have to be forwarded into the following channels, as see in 
https://github.com/d4straub/mag/blob/5b925b16f728ea65cb6fbbbc534b623072abd5a8/main.nf#L365-L368

Reply:
this is correct, you'd still need a if clause to set up the Channels correctly